item-impl/src/main/java/com/example/auction/item/impl/PItemEntity.java

```diff
@@ -103,14 +108,26 @@ private Behavior cancelled(PItemState state) {
         builder.setReadOnlyCommandHandler(GetItem.class, this::getItem);
 
         // Ignore these commands, they may come due to at least once messaging
-        builder.setReadOnlyCommandHandler(UpdatePrice.class,
-                (cmd, ctx) -> ctx.reply(Done.getInstance()));
-        builder.setReadOnlyCommandHandler(FinishAuction.class,
-                (cmd, ctx) -> ctx.reply(Done.getInstance()));
+        builder.setReadOnlyCommandHandler(UpdatePrice.class, this::alreadyDone);
+        builder.setReadOnlyCommandHandler(FinishAuction.class, this::alreadyDone);
 
         return builder.build();
     }
 
+    /**
+     * Convenience method to handle when a command has already been processed (idempotent processing).
+     */
+    private void alreadyDone(Object command, ReadOnlyCommandContext<Done> ctx) {
```

Comment:
Well, if the command defines a ReplyType and the service is waiting on the response ( `ask` ) then replying is a must, right? From the client PoV, it's irrelevant if the message was processed or ignored due to repetition because we're covering a command retrial. Also, this `ignore` method could only be used when the ReplyType is `ReplyType<Done>` (but it's a very common case).

Back to `ctx.done`, I agree with you that we could simply not sent events and keep going.